src/style.css

```diff
@@ -0,0 +1,110 @@
+:root {
+  --star-size: 16px;
+  --distance-between-stars: 4px;
+  --star-top-offset: 2px;
+  --row-offset: 2px;
+}
+
+body {
+  margin: 0;
+}
+
+.stars {
+  display: flex;
+  flex-wrap: wrap;
+}
+
+.stars--0,
+.stars--1,
+.stars--2,
+.stars--3,
+.stars--4,
+.stars--5 {
+  min-width: calc(5 * var(--star-size)
+  + 4 * var(--distance-between-stars) + 2 * var(--row-offset));
+  min-height: var(--star-size);
+  display: flex;
+  justify-content: space-around;
+}
+
+.stars--1::before {
+  width: 16px;
+  height: 100%;
+  background-image: url("images/star-active.svg");
+  background-repeat: no-repeat;
+  background-position: 0 var(--star-top-offset);
+  content: " ";
+}
+
+.stars--2::before {
+  width: calc(2 * var(--star-size) + var(--distance-between-stars));
+  height: 100%;
+  background-image: url("images/star-active.svg"), url("images/star-active.svg");
+  background-repeat: no-repeat, no-repeat;
+  background-position: 0 var(--star-top-offset),
+    calc(var(--star-size) + var(--distance-between-stars))
+    var(--star-top-offset);
+  content: " ";
+}
+
+.stars--3::before {
+  width: calc(3 * var(--star-size) + 2 * var(--distance-between-stars));
+  height: 100%;
+  background-image: url("images/star-active.svg"),
+    url("images/star-active.svg"),
+    url("images/star-active.svg");
+  background-repeat: no-repeat, no-repeat, no-repeat;
+  background-position: 0 var(--star-top-offset),
+    calc(var(--star-size) + var(--distance-between-stars))
+    var(--star-top-offset),
+    calc(2 * var(--star-size) + 2 * var(--distance-between-stars))
+    var(--star-top-offset);
+  content: " ";
+}
+
+.stars--4::before {
+  width: calc(4 * var(--star-size) + 3 * var(--distance-between-stars));
+  height: 100%;
+  background-image: url("images/star-active.svg"),
+    url("images/star-active.svg"),
+    url("images/star-active.svg"),
+    url("images/star-active.svg");
+  background-repeat: no-repeat, no-repeat, no-repeat, no-repeat;
+  background-position: 0 var(--star-top-offset),
+    calc(var(--star-size) + var(--distance-between-stars))
+    var(--star-top-offset),
+    calc(2 * var(--star-size) + 2 * var(--distance-between-stars))
+    var(--star-top-offset),
+    calc(3 * var(--star-size) + 3 * var(--distance-between-stars))
+    var(--star-top-offset);
+  content: " ";
+}
+
+.stars--5 {
+  background-image: url("images/star-active.svg"),
+    url("images/star-active.svg"),
+    url("images/star-active.svg"),
+    url("images/star-active.svg"),
+    url("images/star-active.svg");
+  background-repeat: no-repeat, no-repeat, no-repeat, no-repeat, no-repeat;
+  background-position: var(--row-offset) var(--star-top-offset),
```

Comment:
Oh boi, try to implement it like so:
- position .stars__star correctly on top of each other
- set on .stars__star background-image, and image of regular inactive star
- now based on such rules `.stars--1 > .stars__star:nth-child(-n+1),` set background to active stars